tests/foreman/rhai/test_rhai_client.py

```diff
@@ -69,21 +73,106 @@ def setUpClass(cls):  # noqa
+
+    def test_provision(self):
+        """A new host appears in Red Hat Access Insights inventory
+
+        :id: 0bf2bccf-0973-49fa-923f-7409fba86f85
+
+        :expectedresults: provisioned host is in the inventory
+        """
+        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+            vm.configure_rhai_client(
+                activation_key=self.ak_name,
+                org=self.org_label,
+                rhel_distro=DISTRO_RHEL7)
+            self.assertIn(
+                'Client: 3.',
+                vm.run('insights-client --version').stdout[0],
+            )
+            with Session(self) as session:
+                set_context(session, org=self.org_label)
+                vm_info = session.hosts.get_host_properties(
+                    vm.hostname,
+                    ['Status', 'Operating system', 'Subscription'])
+                self.assertIn(vm_info['Status'], ['Warning', 'OK'])
+                self.assertEqual(vm_info['Subscription'], 'Fully entitled')
+                self.assertTrue(
+                    session.rhai_inventory.search(vm.hostname).is_displayed())
+
+    def test_numeric_group(self):
+        """Check the rule appears when provoked and disappears on Satellite
+        once applied
+
+        :id: 4562e0f9-fff1-4cc7-b7e7-e4779662b3e1
+
+        :expectedresults: rule no more appears on Rules page on portal
+        """
+        message = 'group names are numeric'
+        with VirtualMachine(distro=DISTRO_RHEL7) as vm:
+            vm.configure_rhai_client(
+                activation_key=self.ak_name,
+                org=self.org_label,
+                rhel_distro=DISTRO_RHEL7)
+
+            with Session(self) as session:
+                set_context(session, org=self.org_label)
+
+                session.rhai_inventory.search_and_click(vm.hostname)
+                self.assertIsNotNone(
+                    session.nav.wait_until_element_exists(
+                        locators['insights.rules.modal_window']))
+                self.assertIsNotNone(
+                    session.nav.wait_until_element_exists(
+                        locators['insights.rules.rule_summary']))
+                rules = session.browser.find_elements_by_class_name(
+                    'rule-summary')
+
+                self.assertFalse(
+                    any([message in rule.text for rule in rules]))
+
+                vm.run('groupadd 123456')
+                vm.run('insights-client')
+
+                session.browser.refresh()
+                session.rhai_inventory.search_and_click(vm.hostname)
+                self.assertIsNotNone(session.nav.wait_until_element_exists(
+                    locators['insights.rules.rule_summary']))
+                rules = session.browser.find_elements_by_class_name(
+                    'rule-summary')
+
+                # New rule appeared
+                self.assertTrue(
+                    any(map(lambda rule: message in rule.text, rules)))
```

Comment:
don't we have something specific? I meant whether it is really necessary to use complex statements like this, and if yes, can you please add more details

Reply:
Here we just check if rules have some strings.